testing/scenario_app/bin/run_android_tests.dart

```diff
@@ -311,27 +327,26 @@ Future<void> _run({
       final (Future<int> logcatExitCode, Stream<String> logcatOutput) = getProcessStreams(logcatProcess);
 
       logcatProcessExitCode = logcatExitCode;
+      String? filterToProcessId;
+
       logcatOutput.listen((String line) {
         // Always write to the full log.
         logcat.writeln(line);
 
         // Conditionally parse and write to stderr.
         final AdbLogLine? adbLogLine = AdbLogLine.tryParse(line);
-        switch (adbLogLine?.process) {
-          case null:
-            break;
-          case 'ActivityManager':
-            // These are mostly noise, i.e. "D ActivityManager: freezing 24632 com.blah".
-            if (adbLogLine!.severity == 'D') {
-              break;
-            }
-          // TODO(matanlurey): Figure out why this isn't 'flutter.scenario' or similar.
-          // Also, why is there two different names?
-          case 'utter.scenario':
-          case 'utter.scenarios':
-          case 'flutter':
-          case 'FlutterJNI':
-            log('[adb] $line');
+        if (verbose || adbLogLine == null) {
+          log(line);
+          return;
+        }
+
+        filterToProcessId ??= adbLogLine.tryParseProcess();
```

Comment:
So the intention is the no print until we've parsed a process?  It might be worth adding a comment here for that.

Reply:
Agreed, done.